save-highscores.s

```diff
@@ -0,0 +1,60 @@
+.include "build/tetris.inc"
+.include "ips.inc"
+
+SRAM := $6000
+
+; patch to load from SRAM (and use a length instead of a terminator)
+
+.segment "initHighScoreTable_patch"
+        ips_segment     "initHighScoreTable_patch",$8095
+
+; initHighScoreTable_patch:
+initHighScoreTable:
+        lda SRAM,x
+        cpx #$50
+        beq continueColdBootInit
+        sta highScoreNames,x
+        inx
+        jmp initHighScoreTable
+
+continueColdBootInit:
+
+; start button patch to add saving
+
+.segment "saveHighscoreStartPressed_patch"
+        ips_segment     "saveHighscoreStartPressed_patch",$A337
+
+; saveHighscoreStartPressed_patch:
+        jsr saveHighScores
```

Comment:
this happens every frame, couldn't quite figure out how to do it just when you press start yet

Reply:
Looks like `@flickerStateSelected_checkForStartPressed` is where you want. Probably $A295.